chat_shell/chat_shell/api/v1/response.py

```
@@ -173,315 +208,112 @@ async def _stream_response(
+                    logger.info(
+                        "[RESPONSE] Terminal event sent, marking transport done: "
+                        "task_id=%d, subtask_id=%d, event_type=%s",
+                        task_id,
+                        subtask_id,
+                        event_type,
+                    )
+                    transport.mark_done()
+
+            # Check if chat task completed with error
+            if chat_task.done() and chat_task.exception():
+                raise chat_task.exception()
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)